### PR TITLE
Favorites done

### PR DESCRIPTION
**Pull request by hydroxxxide**
get, add, remove, sort

### CODE REVIEW

**Review by Eldiar004 — APPROVED** [review]
Good

File: src/main/java/com/levante/levante/web/controller/favorite/FavoriteController.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            @Parameter(description = "User ID", required = true)
+            @RequestParam long userId,
+
+            @Parameter(description = "Sort criteria for favorites ('last added', 'price (desc)', 'name')")
+            @RequestParam(required = false) String sortBy,
+
+            @Parameter(description = "Pageable settings for pagination")
+            Pageable pageable
+    ) {
+        return ResponseEntity.ok(favoriteService.getPagesOfFavorite(userId, sortBy, pageable));
```
Comment (i-Coderr) [inline src/main/java/com/levante/levante/web/controller/favorite/FavoriteController.java:45]: Add try catch

File: src/main/java/com/levante/levante/web/controller/favorite/FavoriteController.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            @Parameter(description = "User ID", required = true)
+            @RequestParam long userId,
+
+            @Parameter(description = "Product ID")
+            @RequestParam long productId,
+
+            @Parameter(description = "Pageable settings for pagination")
+            Pageable pageable
+    ) {
+        return ResponseEntity.ok(favoriteService.addOrRemoveFromFavorites(userId, productId, pageable));
```
Comment (i-Coderr) [inline src/main/java/com/levante/levante/web/controller/favorite/FavoriteController.java:68]: Add try catch

**Review by i-Coderr — COMMENTED** [review]
@hydroxxxide comments

**Pull request merged**